driver-ui/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/AbstractWebDriverRequest.java

```diff
@@ -42,4 +44,27 @@ public DesiredCapabilities getDesiredCapabilities() {
         }
         return desiredCapabilities;
     }
+
+    /**
+     * Cloning of DesiredCapabilites with SerializationUtils occurs org.apache.commons.lang3.SerializationException: IOException while reading or closing cloned object data
+     * -> We have to backup the current caps and clone WebDriverRequest without caps. After cloning the original caps are added again.
+     * -> Caps can cloned via merge() method.
+     *
+     * @return
+     */
```

Comment:
imo we should not implement / use clone() here at all. The original bug is the result of a misuse of clone, if we can, we should fix the root cause and remove clone altogether here.
See https://www.artima.com/articles/josh-bloch-on-design#part13 as one example why clone should almost always not be used. There are more arguments online.

Reply:
There is in fact not a misuse of clone. The article points out in "Copy Constructor vs. Cloning" that the `Clonable` interface doesn't provide any method and therefore objects cannot be casted and called on `clone()`. Clonable objects need to implement `clone()` as public method and are therefore not reliable. I see that there could be a design flaw but that argument is not strong enough to prevent that standard way of implementing cloning in Java.

Reply:
@martingrossmann i will approve if you remove Serializable Interface from AbstractWebDriverRequest. 
Doing this "proper" without clone is probably out of scope for this bugfix, unfortunately.

two reasons why to remove Serializable:

**1. AbstractWebDriverRequest is not actually Serializable** 
because org.openqa.selenium.DesiredCapabilities apparently does not implement Serializable in the way that org.apache.commons.lang3.SerializationUtils expects it. 

I found a stackoverflow article that pointed me to it : https://stackoverflow.com/questions/9747443/java-io-invalidclassexception-no-valid-constructor/51492340
I debugged this to verify stackoverflow article claims and found the following : 

I think org.openqa.selenium.AbstractCapabilities would need to implement a public constructor without parameters for DesiredCapabilities to actually be Serializable. This seems to be an implicit requirement of the implementations of java.io.ObjectStreamClass and java.io.ObjectOutputStream - in java 11.0.8. The constructor of org.openqa.selenium.AbstractCapabilities only has default access modifier, it is only accessible from package local functions. I think this might trigger the exception org.apache.commons.lang3.SerializationException.

If I am not mistaken this is a bug in Selenium - I will open an issue with them. Maybe they just don't want to support this specific way of serializing DesiredCapabilities, but different ways would be supported.. let's find out.

**2. a consumer can extend AbstractWebDriverRequest and add non-serializable fields** 
that result in org.apache.commons.lang3.SerializationException or possibly even stall execution - you cannot promise to a consumer of Testerra API that anything he adds is serializable. Ideas how to solve this via architecture change in comment below.

Reply:
@mreiche on cloning as bad practice :
I extract from Bloch's texts that cloning is deeply flawed and should be avoided at almost all costs - implicit: by changing architecture accordingly. Imo this should be considered bad practice since at least 2002 (Effective Java), warranting the word "misuse" from my pov - but we can leave the discussion at that, it is an opinionated issue for sure.
Imo clone can and should be avoided by changing architecture- but this it out of scope for this bugfix. see my comment directly on the issue.

Reply:
@gogro All I read about not using `Cloneable` is [1], that the main reason why you should avoid using it, because it doesn't provide the `clone()` method. Which is in fact a flaw or maybe "broken" but the workarround is to just implement it. Besides that, cloning objects works fine.

I dont think that copy constructors or factory methods are a better solution since the concrete implementation of `WebDriverRequest` is not known at this point we request a copy.

I agree that we should also shoud avoid using `Serializable` and just call `clone()` instead.

[1]: https://howtodoinjava.com/java/cloning/cloneable-interface-is-broken-in-java/

Reply:
> the concrete implementation of WebDriverRequest is not known at this point we request a copy
exactely, my idea is : we should consider making the concrete implementation known in public API, in the future. That is the point in https://github.com/telekom/testerra/pull/98#pullrequestreview-719569125